Bug 1865930: Fix runtime error on project details page

TheRealJon opened the pull request:
In area chart component, filter empty sets from the `data` prop before using it to calculate legend data or render a chart.

spadgett commented on frontend/public/components/graphs/area.tsx:
```diff
@@ -91,9 +91,9 @@ export const AreaChart: React.FC<AreaChartProps> = ({
-        symbol: d[0].symbol,
+        childName: d[0]?.description,
+        name: d[0]?.description,
+        symbol: d[0]?.symbol,
```
Do we know if it's safe to pass `undefined` for these values? How does it impact the chart when this happens?

TheRealJon replied:
I followed this down through the ChartLegendTooltip component and it looks like it should be okay if these are undefined. I'm not 100% sure what effect this would have on the chart since in this case `d[0] === undefined` means there is an empty data set. Let me dig a little bit more.

TheRealJon commented:
/hold

Testing implications of this

TheRealJon commented:
/hold cancel

@spadgett Looked a little more deeply at this and I think solving the 'empty data' set issue before even getting to the legend logic makes more sense. I added logic to filter the `data` prop down to only data sets that have data points and removed the optional chaining from my previous solution.

spadgett commented:
/hold
for queue fix #6225

spadgett commented:
/hold cancel

spadgett commented:
/retest
